### PR TITLE
Make all decomps register to Meta, and test it

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #77477
* __->__ #77386

Added an API to let you query if there's already a registration for a function, and then just made `register_decomposition` unconditionally register the meta implementation if there is not already one in core. You can disable it with `disable_meta`.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -14460,7 +14461,7 @@ def generate_std_var_kwargs(t: torch.Tensor, **kwargs):
            supports_forward_ad=True,
            supports_fwgrad_bwgrad=True,
            gradcheck_wrapper=lambda *args, **kwargs: gradcheck_wrapper_triangular_input(*args, idx=1, **kwargs),
-           decorators=[skipCUDAIfNoMagma],
+           decorators=[skipCUDAIfNoMagma, skipCPUIfNoLapack],
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:14464]: Nice update

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -14705,6 +14706,7 @@ def generate_std_var_kwargs(t: torch.Tensor, **kwargs):
            dtypes=floating_and_complex_types(),
            sample_inputs_func=sample_inputs_linalg_solve_triangular,
            supports_fwgrad_bwgrad=True,
+           skips=(skipCPUIfNoLapack,),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:14709]: Nice skip

**Review by mruberry — APPROVED** [review]
LGTM!

File: test/test_meta.py [hunk 3]
```diff
@@ -331,7 +332,6 @@
     torch.nn.functional.adaptive_avg_pool2d,  # MISSING aten::_adaptive_avg_pool2d
     torch.nn.functional.adaptive_avg_pool3d,  # MISSING aten::_adaptive_avg_pool3d
     torch.nn.functional.batch_norm,  # MISSING aten::native_batch_norm
-    torch.nn.functional.binary_cross_entropy,  # MISSING aten::binary_cross_entropy
```
Comment (Chillee) [inline test/test_meta.py:334]: Are the `# missing` parts autogenerated btw?

**Comment by Chillee** [comment]
I'm a little ... surprised the tests all pass? Most of the decomps haven't had that much attention paid to strides so far. Perhaps the set of ops for which we've written decompositions mostly have trivial stride propagation.

File: test/test_meta.py (hunk 3 above)
Comment (ezyang) [reply test/test_meta.py:334]: yup

**Comment by ezyang** [comment]
> I'm a little ... surprised the tests all pass? Most of the decomps haven't had that much attention paid to strides so far. Perhaps the set of ops for which we've written decompositions mostly have trivial stride propagation.

There's a testing problem I'm working on, which is that no backwards is being tested for meta.

**Comment by ezyang** [comment]
folded into https://github.com/pytorch/pytorch/pull/77477

**Pull request closed without merging**